lib/util/fix-tracker.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @fileoverview Helper class to aid in constructing fix commands.
+ * @author Alan Pierce
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+/**
+ * A helper class to combine fix options into a fix command. Currently, it
+ * exposes some "retain" methods that extend the range of the text being
+ * replaced so that other fixes won't touch that region in the same pass.
+ */
+class FixTracker {
```

Comment:
Yep, agreed that putting this stuff on `ruleFixer` would be cleaner (assuming `ruleFixer` can be changed to be aware of the source code), except for the fact that it's a public API, so we need to be careful about changing it.